src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -4303,7 +4303,9 @@ namespace ZoneEquipmentManager {
 					// update the return air node for zonal and central on/off systems
 					Node( ReturnNode ).Temp = Node( ZoneNode ).Temp;
 				}
-
+				if ( Zone( ActualZoneNum ).HasApproachTempToReturnAir && !( DataGlobals::BeginSimFlag ) ) {
+					Node( ReturnNode ).Temp = Zone( ActualZoneNum ).AdjustedTempToReturnAir;
+				}
```

Comment:
This stomps on any other return air heat gain terms.  This should be moved up earlier (around line 4244) to reset TempZoneAir and TempRetAir.  Then other return air heat gain terms can be added to the base return temp from the IT Equipment.  Either that, or you need to check for all other possible return air heat gains (lights, airflow windows, refrigeration equipment, others?) and throw a fatal error is any of them are used in the same zone with IT equipment using the return approach method.